tests/general/GlbLubTest.java

```diff
@@ -0,0 +1,113 @@
+package general;
+
+import org.checkerframework.checker.crypto.qual.AllowedAlgorithms;
+import org.checkerframework.checker.crypto.qual.AllowedProviders;
+import org.checkerframework.checker.crypto.qual.Bottom;
+import org.checkerframework.checker.crypto.qual.UnknownAlgorithmOrProvider;
+
+class GlbLubTest {
+
+    void testLub1(
+            @AllowedAlgorithms({"algorithm1"}) String algo,
+            @AllowedProviders({"provider1"}) String prov) {
+
+        String either;
+        if (true) {
+            either = algo;
+        } else {
+            either = prov;
+        }
+        // :: error: assignment
+        algo = either;
+        // :: error: assignment
+        prov = either;
+    }
+
+    void testLub2(
+            @AllowedAlgorithms({"algorithm1", "algorithm2"}) String algo1,
+            @AllowedAlgorithms({"algorithm1", "algorithm3"}) String algo2) {
+
+        String either;
+        if (true) {
+            either = algo1;
+        } else {
+            either = algo2;
+        }
+        // :: error: assignment
```

Comment:
You will need to update these error keys again.

Reply:
Thanks, fixed!